src/main/java/MealDemo/rest/MealPlanningRESTController.java

```diff
@@ -0,0 +1,66 @@
+package MealDemo.rest;
+
+import MealDemo.entity.Frequency;
+import MealDemo.entity.Meal;
+import MealDemo.service.MealPlanningServiceImpl;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/test")
+public class MealPlanningRESTController {
+
+    private MealPlanningServiceImpl mealPlanningServiceImpl;
+
+    public MealPlanningRESTController(MealPlanningServiceImpl mealPlanningServiceImpl) {
+        this.mealPlanningServiceImpl = mealPlanningServiceImpl;
+    }
+
+
+
+    @GetMapping("/viewM")
```

Comment:
For this path `viewM` and `viewF` a better more restful path would be:

```
@GetMapping("/meals")

and 

@GetMapping("/meals/frequencies")
```

This is better design because:

- The naming is clear. You don't need to guess what M/F means at the end
- You can expand both of these to later take into account query parameter so you can apply filtering logic for example you could have

```
/meals?category=chicken
```

What this means is that using the same endpoint you have optional filtering based on parameters. You don't need to do this now but it's good practice to design these based endpoints in this manner so they can be extended later if needed. For now I would just rename the paths to 

```
@GetMapping("/meals")
@GetMapping("/meals/frequencies")
```